### PR TITLE
NEXT-35968 - Fix OpenApi Schema validation

### PR DESCRIPTION
**Pull request by ghost**
### 1. Why is this change necessary?
- Currently the github workflow for the OpenApi Schema validation fails because it's not setup correctly
- Also the action fails because a few api specs in the core a wrong

### 2. What does this change do, exactly?
* Changed `01-lint-openapi` to run correctly & cache composer files
* Changed `CalculatedPrice` API Schema to correctly use `oneOf` with `null`
* Changed `Product` API Schema to correctly use `oneOf` with `null`
* Changed `ReferencePrice` API Schema to correctly use `oneOf` with `null`

### 3. Describe each step to reproduce the issue or behaviour.
- Run the OpenApi Schema validation github action

### 4. Please link to the relevant issues (if any).
- /

### 5. Checklist

- [X] I have rebased my changes to remove merge conflicts
- [X] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [X] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by ghost** [comment]
@philipreinken I fixed the GitHub action to correctly run and changed it to match the structure of the other actions.

**Comment by BrocksiNet** [comment]
@mkucmus, can you please validate the changes made to OpenAPI Schema? Thx.

File: changelog/_unreleased/2024-04-26-fix-openapi-validation.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+issue: NEXT-00000
+author: Benjamin Wittwer
+author_email: benjamin.wittwer@a-k-f.de
+author_github: akf-bw
+---
+# Core
+* Changed `01-lint-openapi` to run correctly & cache composer files
+* Changed `CalculatedPrice` API Schema to correctly use `oneOf` with `null`
+* Changed `Product` API Schema to correctly use `oneOf` with `null`
+* Changed `ReferencePrice` API Schema to correctly use `oneOf` with `null`
```
Comment (mkucmus) [inline changelog/_unreleased/2024-04-26-fix-openapi-validation.md:12]: thank you for this 💙 
from now on, we will have `calculatedPrice: calculatedPrice | null` and so on for the entities which are allowed to be null (and also present in the response).

@shyim could you have a look at the CI improvement as well?

**Review by mkucmus — APPROVED** [review]
From the schema perspective, `nullable` option is set correctly 👍🏼

**Comment by leduc92** [comment]
Your merge request has been merged @akf-bw 🎉  , thanks for your contribution.

**Pull request closed without merging**